src/core/core.cpp

```diff
@@ -735,6 +735,13 @@ void wf::compositor_core_impl_t::erase_view(wayfire_view v)
     views.erase(it);
 }
 
+void wf::compositor_core_impl_t::show_window_menu(wayfire_view view)
```

Comment:
Why do we need a function in core?

Reply:
This is because other things can call this, for example foreign-toplevel could have a show_window_menu request, so that a menu can be shown in response to right-clicking on a button in the window-list of wf-panel, instead of the gtk menu that is there currently, which is suboptimal. Another use-case could be that the window-menu plugin could detect click-on-desktop to open a menu as well.

Reply:
We could emit the signal directly and same for foreign-toplevel, then for other things like click-on-desktop menu, we can just run() a menu and match it by app_id.

Reply:
Yes, I think we should emit the signal directly where applicable (note that plugins could emit this too, for click-on-desktop)